testproject/Assets/GlobalGameState/Scripts/InGameManager.cs

```diff
@@ -0,0 +1,444 @@
+    /// Executed once per update when the m_InGameState.Value is InGameStates.Waiting
+    /// You could animate something here while players wait
+    /// </summary>
+    void WaitingUpdate()
+    {
+        //Waiting update here
+    }
+
+    /// <summary>
+    /// PlayingUpdate
+    /// Executed once per update when the m_InGameState.Value is InGameStates.Playing
+    /// All in game logic happens here
+    /// </summary>
+    void PlayingUpdate()
+    {
+        //Game update here
+    }
+
+    /// <summary>
+    /// PausedUpdate
+    /// Executed once per update when the m_InGameState.Value is InGameStates.Paused
+    /// You could animate something here while the game is paused
+    /// </summary>
+    void PausedUpdate()
+    {
+        //Paused update here
+    }
+
+    /// <summary>
+    /// OnExitingGame
+    /// Executed once per update when the m_InGameState.Value is InGameStates.Exiting
+    /// </summary>
+    void OnExitingGame()
+    {
+        //Server is authoritative for the exiting timer
+        if (IsServer)
+        {
+            m_ExitingTime.Value = Mathf.Max(0, m_ExitingTime.Value - Time.deltaTime);
+        }
+
+        //If no time is left, then exit
+        if (m_ExitingTime.Value == 0)
+        {
+            //Server waits for no more players, and then exits the game
+            if (IsServer && NetworkingManager.Singleton.ConnectedClientsList.Count <= 1)
+            {
+                //De-register from our events
+                GlobalGameState.Singleton.allPlayersLoadedScene -= AllPlayersLoadedScene;
+                NetworkingManager.Singleton.OnClientDisconnectCallback -= OnClientDisconnectCallback;
+                GlobalGameState.Singleton.SetGameState(GlobalGameState.GameStates.ExitGame);
+            }
+            else if (!IsServer)
+            {
+                //Clients always exit immediately
+                GlobalGameState.Singleton.SetGameState(GlobalGameState.GameStates.ExitGame);
+            }
+        }
+        else
+        {
+            m_ExitingGame.text = "Exiting in " + m_ExitingTime.Value.ToString() + " seconds!";
+        }
+    }
+
+    /// <summary>
+    /// OnExitGame
+    /// Tied to the "X" button in the top right corner of the InGame Scene
+    /// </summary>
+    public void OnExitGame()
+    {
+        if (IsServer)
+        {
+            m_InGameState.Value = InGameStates.Exiting;
+        }
+        else
+        {
+            //Clients always exit immediately
+            GlobalGameState.Singleton.SetGameState(GlobalGameState.GameStates.ExitGame);
+        }
+    }
+
+    public void OnNextLevel()
+    {
+        if(IsServer)
+        {
+            GlobalGameState.Singleton.SetGameState(GlobalGameState.GameStates.InGame);
+        }
+    }
+
+    /// <summary>
+    /// ServerCommandInputUpdate
+    /// This is where you can add commands or detect key presses to perform server/host operations
+    /// </summary>
+    void ServerCommandInputUpdate()
```

Comment:
Command might confuse some people because UNet/Mirror use Command to refer to ServerRPCs